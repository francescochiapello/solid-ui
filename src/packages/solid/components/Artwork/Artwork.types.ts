export interface ArtworkStyleProperties {
  /** URI of fallback image to be shown if the src request fails */
  fallbackSrc?: NodeStyles['src'];

  /**
   * solid color background, displayed if no src is provided or URI is invalid
   *
   * core reference: {@link https://github.com/lightning-js/renderer/blob/aefee0064a48055b3cf8dd459396ad4996b68ef5/src/main-api/INode.ts#L121 INodeWritableProps.color}
   */
  fillColor?: NodeStyles['color'];

  /**
   * starting color the linear gradient effect to apply on top of image
   *
   * core reference: {@link https://github.com/lightning-js/renderer/blob/aefee0064a48055b3cf8dd459396ad4996b68ef5/src/core/renderers/webgl/shaders/effects/LinearGradientEffect.ts#L35 LinearGradientEffectProps.colors}
   * */
  gradientColor?: NodeStyles['linearGradient'];

  /**
   * The scale value multiplies the provided width and height of the Node.
   *
   * core reference:
   * {@link https://github.com/lightning-js/renderer/blob/aefee0064a48055b3cf8dd459396ad4996b68ef5/src/main-api/INode.ts#L260 INodeWritableProps.scale}
   *
   */
  imageScale?: NodeStyles['scale'];

  /**
   * X position of the Node's Pivot Point
   *
   * core reference:
   * {@link https://github.com/lightning-js/renderer/blob/aefee0064a48055b3cf8dd459396ad4996b68ef5/src/main-api/INode.ts#L375 INodeWritableProps.pivotX}
   */
  imageScalePivotX?: NodeStyles['pivotX'];

  /**
   * Y position of the Node's Pivot Point
   *
   * core reference:
   * {@link https://github.com/lightning-js/renderer/blob/aefee0064a48055b3cf8dd459396ad4996b68ef5/src/main-api/INode.ts#L390 INodeWritableProps.pivotY}
   */
  imageScalePivotY?: NodeStyles['pivotY'];
  radius?: NodeStyles['borderRadius'];

  // TODO future adds
  // blur: number; not in renderer yet
  // padding: number; is this just support for CSS padding?
}

export interface ArtworkStyles {
  tone: Tone;
  Container: NodeStyleSet<{
    fallbackSrc: NodeStyles['src'];
    fillColor: NodeStyles['color'];
    gradientColor: NodeStyles['color'];
  }>;
}

export type ArtworkConfig = ComponentStyleConfig<ArtworkStyleProperties>;
