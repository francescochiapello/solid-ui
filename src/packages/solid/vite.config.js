/*
 * Copyright 2023 Comcast Cable Communications Management, LLC
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { defineConfig } from 'vite';
import solidPlugin from 'vite-plugin-solid';
import { importChunkUrl } from '@lightningjs/vite-plugin-import-chunk-url';
import path from 'path';

const config = {
  optimizeDeps: {
    include: [],
    exclude: [
      '@lightningjs/solid',
      '@lightningjs/solid-primitives',
      '@lightningjs/renderer/core',
      '@lightningjs/renderer/workers/renderer'
    ]
  },
  plugins: [
    solidPlugin({
      solid: {
        moduleName: '@lightningjs/solid',
        generate: 'universal'
      }
    }),
    importChunkUrl()
  ],
  base: './',
  build: {
    lib: {
      fileName: 'index',
      entry: './index.ts',
      formats: ['es']
    },
    rollupOptions: {
      external: ['theme', 'solid-js', '@lightningjs/solid', '@lightningjs/solid-primitives']
    },
    minify: false,
    sourcemap: true,
    outDir: './dist',
    target: 'esnext',
    polyfillDynamicImport: false
  },
  resolve: {
    alias: {
      theme: path.resolve(__dirname, '../l3-ui-theme-base/theme.js'),
      utils: path.resolve(__dirname, '../../shared/utils/index.ts')
    },
    dedupe: ['solid-js', '@lightningjs/solid']
  },
  server: {
    hmr: false,
    headers: {
      'Cross-Origin-Opener-Policy': 'same-origin',
      'Cross-Origin-Embedder-Policy': 'require-corp'
    }
  },
  test: {
    /**
     * browser mode causes tests to fail with the following error:
     * SyntaxError: The requested module 'ui-components/node_modules/.pnpm/chalk@3.0.0/node_modules/chalk/source/index.js?v=6d2cbdb3' does not provide an export named 'default'`
     */
    enabled: false,
    headless: false,
    provider: 'playwright',
    name: 'chromium',
    slowHijackESM: false,
    deps: {
      inline: [/solid-js/, /solid-testing-library/],
      optimizer: {
        web: {
          include: ['@lightningjs/solid', '@lightningjs/renderer']
        }
      }
    },
    transformMode: { web: [/\.[jt]sx?$/] },
    globals: true,
    resolve: {
      conditions: ['development', 'browser']
    }
  },
  publicDir: '../../shared/public'
};

export default defineConfig(config);
