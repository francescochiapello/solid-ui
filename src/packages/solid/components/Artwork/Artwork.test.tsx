/**
 * Copyright 2024 Comcast Cable Communications Management, LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import Artwork from './Artwork.jsx';
import { render } from '../../test-utils.jsx';
import { describe, expect, it } from 'vitest';
import { hexColor } from '@lightningjs/solid';

describe('Artwork', () => {
  it('renders', async () => {
    const tree = await render(() => <Artwork color={hexColor('#ffffff')} />);
    expect(tree).toMatchSnapshot();
    // expect(true).toBe(true);
  });
});
