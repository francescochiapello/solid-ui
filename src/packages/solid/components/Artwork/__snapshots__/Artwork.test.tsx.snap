// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Artwork > renders 1`] = `
ElementNode {
  "_animationQueue": [],
  "_animationRunning": false,
  "children": Children [
    ElementNode {
      "_animationQueue": [],
      "_animationRunning": false,
      "_effects": {
        "radius": {
          "radius": 8,
        },
      },
      "_parent": [Circular],
      "_queueDelete": false,
      "_style": {
        "borderRadius": 8,
        "disabled": {},
        "fallbackSrc": undefined,
        "fillColor": "0x181819b3",
        "focus": {},
        "gradientColor": "0x181819ff",
        "pivotX": 0.5,
        "pivotY": 0.5,
        "scale": undefined,
      },
      "children": Children [],
      "disabled": {},
      "fallbackSrc": undefined,
      "fillColor": "0x181819b3",
      "focus": {},
      "gradientColor": "0x181819ff",
      "lng": {
        "color": "0x181819b3",
        "pivotX": 0.5,
        "pivotY": 0.5,
        "scale": undefined,
      },
      "rendered": false,
      "type": 0,
    },
  ],
  "lng": {},
  "onMount": [Function],
  "rendered": false,
  "type": 0,
}
`;
